portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -24,9 +27,32 @@
 @WebServlet("/data")
 public class DataServlet extends HttpServlet {
 
+  ArrayList<String> messages;
+
+  public DataServlet() {
+    messages = new ArrayList<>();
+  }
+
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
     response.setContentType("text/html;");
-    response.getWriter().println("<h1>Hello, Joshua!</h1>");
+    response.getWriter().println(convertToJson(messages));
+
+  }
+
+  @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    String commentData = getNewComment(request);
+    messages.add(commentData);
+    response.sendRedirect("/index.html");
+  }
+
+  private String convertToJson(ArrayList<String> messages) {
+    Gson gson = new Gson();
+    return gson.toJson(messages);
+  }
+
+  private String getNewComment(HttpServletRequest request) {
+    return request.getParameter("comment-input");
   }
```

Comment:
Consider simply calling request.getParameter above instead of creating a helper function to do it. 

This post (see top answer in particular) has a good discussion about when to make functions containing only 1 line, and when to not:
https://softwareengineering.stackexchange.com/questions/107669/one-line-functions-that-are-called-only-once

Reply:
Yeah, I had plans to expand the function to do more things (in which case, it would have been void instead), but I realized that that might not happen. You'll probably see in the changes for the next part that I accidentally pushed, I changed that.